src/engine/QueryExecutionTree.cpp

```diff
@@ -35,25 +35,23 @@ string QueryExecutionTree::asString(size_t indent) {
   for (size_t i = 0; i < indent; ++i) {
     indentStr += " ";
   }
-  if (_asString.size() == 0) {
-    if (_rootOperation) {
-      std::ostringstream os;
-      os << indentStr << "{\n"
-         << _rootOperation->asString(indent + 2) << "\n"
-         << indentStr << "  qet-width: " << getResultWidth() << " ";
-      if (LOGLEVEL >= TRACE && _qec) {
-        os << " [estimated size: " << getSizeEstimate() << "]";
-        os << " [multiplicities: ";
-        for (size_t i = 0; i < getResultWidth(); ++i) {
-          os << getMultiplicity(i) << ' ';
-        }
-        os << "]";
+  if (_rootOperation) {
+    std::ostringstream os;
+    os << indentStr << "{\n"
+       << _rootOperation->asString(indent + 2) << "\n"
+       << indentStr << "  qet-width: " << getResultWidth() << " ";
+    if (LOGLEVEL >= TRACE && _qec) {
+      os << " [estimated size: " << getSizeEstimate() << "]";
```

Comment:
After a tree was cached a new call would give new (exact) size estimates so this part would change. So I think with higher `LOGLEVEL` the optimization doesn't really work? Maybe these estimates really shouldn't be in the cache string. So maybe we should put this part behind a flag parameter that's only used when printing to the console? Then we can also print this at every `LOGLEVEL`

Reply:
You are right. For now I left it since I would suggest putting the whole cache-key vs nice formatting business in a different PR since I want to do this in a clean way.